intr.S

```diff
@@ -0,0 +1,91 @@
+    .set noreorder // Disable automatic instruction reordering
+
+    /* Interrupts vector */
+    .section .exception
+    .globl _interrupt_handler
+    .org 0x200
+_interrupt_handler:
+#define REG_STACK_STORE_SIZE 76
+    /* Please, no nested interrupts for now. */
+    di
+
+    /* Allocate space for register storage on the stack. */
+    subu $sp, $sp, REG_STACK_STORE_SIZE
+
+    /* Save registers on the stack. Saving all t,a and v
+       registers, as well as LO and HI.
+       k registers do not require saving.
+       s and sp registers are saved by compiler-generated
+       procedure entry, if the procedure modifies them.*/
+    sw $t0,  0 ($sp)
+    sw $t1,  4 ($sp)
+    sw $t2,  8 ($sp)
+    sw $t3, 12 ($sp)
+    sw $t4, 16 ($sp)
+    sw $t5, 20 ($sp)
+    sw $t6, 24 ($sp)
+    sw $t7, 28 ($sp)
+    sw $t8, 32 ($sp)
+    sw $t9, 36 ($sp)
+
+    /* Asking early, as these instructions require a few cycles. */
```

Comment:
Apparently I have confused MIPS-II with MIPS32-II, there is indeed no restriction on the instructions surrounding `mflo`/`mfhi`.